src/sas/qtgui/Plotting/Plotter2D.py

```diff
@@ -396,6 +422,47 @@ def setSlicer(self, slicer, reset=True):
         # Clear current slicer
         if self.slicer is not None:
             self.slicer.clear()
+
+        # Clear the old slicer plots so they don't reappear later
+        if hasattr(self, '_item'):
+            item = self._item
+            if self._item.parent() is not None:
+                item = self._item.parent()
+
+            # Go through all items and see if they are a plot. The checks done here are not as thorough
+            # as GuiUtils.deleteRedundantPlots (which this takes a lot from). Will this cause problems?
+            # Primary concern is the check (plot_data.plot_role == DataRole.ROLE_DELETABLE) as I don't
+            # know what it does. The other checks seem to be related to keeping the new plots for that function
```

Comment:
Why reinvent the wheel? What advantage does this have over the existing method?

Reply:
I did it this way because I would have to provide GuiUtils.deleteRedundantPlots a list of all of the plots to keep. I was unsure of a way to reliably include all other plots to keep and exclude only the old slicers. If someone has a good idea of what all of the plots that could possibly be a child of the 2D data then they could make use of GuiUtils.deleteRedundantPlots and delete the addition here.

Reply:
Talked with @butlerpd and I won't have time to get this change in for the pull request. I've created #2838 to address this at some point.